helpers/network_device.rb

```diff
@@ -28,11 +28,31 @@ def network_devices
   def create_net(net_hash)
     OpenStruct.new(net_hash).tap do |o|
       o.speed = format_bits_value((net_hash['capacity'] || net_hash['size'] || 0).to_i)
+
+      # Set speed for Virtual & Infiniband interfaces
+      if o.speed == '0 bit/s' and o.logicalname&.match? /ib/
+        o.speed = 'Infiniband'
+      elsif o.speed == '0 bit/s'
+        o.speed = 'Virtual'
+      end
+
+      # Handle logicalname for virtio devices (as information is buried in 'node' hash)
+      if o.logicalname.nil?
+        o.logicalname = o.node['logicalname']
+      end
+
+      # Handle mac address for virtio devices
+      if o.serial.nil?
+        o.serial = o.node['serial']
+      end
     end
   end
   network_devices = []
   find_hashes_with_key_value(@asset_hash, 'class', 'network')&.each do |net|
-    network_devices << create_net(net)
+    # Ignore virtual interfaces (bridges, tunnels and bonds)
+    unless find_hashes_with_key_value(net['configuration']['setting'], 'id', 'driver$')[0]['value'].match? /bridge|tun|bonding/
```

Comment:
Similar here with the `find_hashes_with_key_value`.

Reply:
Again, same as above regarding consistency of the hash (as it's based on command output) and that any changes should cause errors otherwise we may miss it!